File: app/src/main/java/tech/pauly/findapet/shared/datastore/DiscoverToolbarTitleUseCase.java
package tech.pauly.findapet.shared.datastore;

import android.support.annotation.StringRes;

public class DiscoverToolbarTitleUseCase implements UseCase {

    @StringRes
    private int title;

    public DiscoverToolbarTitleUseCase(@StringRes int title) {
        this.title = title;
    }

    @StringRes
    public int getTitle() {
        return title;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }

        DiscoverToolbarTitleUseCase that = (DiscoverToolbarTitleUseCase) o;

        return title == that.title;
    }

    @Override
    public int hashCode() {
        return title;
    }
}

File: app/src/main/java/tech/pauly/findapet/shared/datastore/DiscoverToolbarTitleUseCase.kt
package tech.pauly.findapet.shared.datastore

import androidx.annotation.StringRes

open class DiscoverToolbarTitleUseCase(@StringRes open val title: Int) : UseCase {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as DiscoverToolbarTitleUseCase

        if (title != other.title) return false

        return true
    }

    override fun hashCode(): Int {
        return title
    }
}